File: backend/models/DemoSession.js
const mongoose = require("mongoose");

const demoSessionSchema = new mongoose.Schema(
  {
    matchId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Match",
      required: true
    },

    // REQUIRED BASED ON CONTROLLER
    studentId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true
    },

    teacherId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true
    },

    requestedBy: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User"
    },

    scheduledAt: {
      type: Date
    },

    status: {
      type: String,
      enum: ["requested", "pending", "approved", "rejected", "completed"],
      default: "requested"
    }
  },
  { timestamps: true }
);

module.exports = mongoose.model("DemoSession", demoSessionSchema);

File: lib/src/core/models/demo_session.dart
class DemoSession {
  final String id;
  final String matchId;
  final String studentId;
  final String teacherId;
  final String status;
  final DateTime? scheduledAt;

  DemoSession({
    required this.id,
    required this.matchId,
    required this.studentId,
    required this.teacherId,
    required this.status,
    required this.scheduledAt,
  });

  factory DemoSession.fromJson(Map<String, dynamic> json) {
    return DemoSession(
      id: json["_id"] ?? "",
      matchId: json["matchId"] ?? "",
      studentId: json["studentId"] ?? "",
      teacherId: json["teacherId"] ?? "",
      status: json["status"] ?? "",
      scheduledAt: json["scheduledAt"] != null
          ? DateTime.parse(json["scheduledAt"])
          : null,
    );
  }
}
